x/vm/keeper/state_transition.go

```diff
@@ -350,14 +382,32 @@ func (k *Keeper) ApplyMessage(ctx sdk.Context, msg core.Message, tracer *tracing
 // # Commit parameter
 //
 // If commit is true, the `StateDB` will be committed, otherwise discarded.
-func (k *Keeper) ApplyMessageWithConfig(ctx sdk.Context, msg core.Message, tracer *tracing.Hooks, commit bool, cfg *statedb.EVMConfig, txConfig statedb.TxConfig, internal bool) (*types.MsgEthereumTxResponse, error) {
+func (k *Keeper) ApplyMessageWithConfig(
+	ctx sdk.Context,
+	msg core.Message,
+	tracer *tracing.Hooks,
+	commit bool,
+	cfg *statedb.EVMConfig,
+	txConfig statedb.TxConfig,
+	internal bool,
+	overrides *rpctypes.StateOverride,
```

Comment:
@vladjdk this functions is starting to have a complicated signature. do you think we could move more of these options into configs etc?